src/IECoreScene/CurvesAlgoUpdateEndpointMultiplicity.cpp

```diff
@@ -162,28 +126,10 @@ struct DuplicateEndPoints
 		const std::vector<T> &in = data->readable();
 		typename S<std::vector<T>>::Ptr newOut = new S<std::vector<T>>();
 
-		if( primVar.interpolation == PrimitiveVariable::Vertex )
-		{
-			if( m_expand )
-			{
-				expandVertex( in, newOut->writable(), curves, primVar, m_targetBasis );
-			}
-			else
-			{
-				compressVertex( in, newOut->writable(), curves, primVar, m_targetBasis );
-			}
-		}
-		else if( primVar.interpolation == PrimitiveVariable::Varying || primVar.interpolation == PrimitiveVariable::FaceVarying )
-		{
-			if( m_expand )
-			{
-				expandVarying( in, newOut->writable(), curves, primVar, m_targetBasis );
-			}
-			else
-			{
-				compressVarying( in, newOut->writable(), curves, primVar, m_targetBasis );
-			}
-		}
+
+		int adjust = primVar.interpolation == PrimitiveVariable::Vertex ? m_adjustVertex : m_adjustVarying;
```

Comment:
Isn't this going to attempt to adjust Uniform and Constant primvars as though they were Varying?

Reply:
We only run EndpointDuplicator on Vertex and Varying primvars.  Do you want an assert or something?